packages/preview-middleware/templates/flp/sandbox.html

```diff
@@ -63,7 +63,7 @@
                 sap.ushell.Container.getServiceAsync('AppLifeCycle').then((serviceInstance) => {
                     serviceInstance.attachAppLoaded((event) => {
                         var oView = event.getParameter('componentInstance');
-                        sap.ui.require(["sap/ui/rta/api/startAdaptation"], async function (startAdaptation) {
+                        sap.ui.require(["sap/ui/rta/api/startAdaptation"<% if (locals.flex?.pluginModule) { %>, "<%- flex.pluginModule %>"<% } %>], async function (startAdaptation<% if (locals.flex?.pluginModule) { %>, loadPlugin<% } %>) {
```

Comment:
In general, maybe we should introduce some general if statements and use those to differentiate between the different preview modes, hence normal, variant management, adaptation? We will have code duplication in the template, but the template will be more readable? Just an idea, what do you think?

Reply:
What do you have in mind? I need an example to better understand it.

Reply:
This is what I have in mind
```JS
   <script>
    case 'NORMAL PREVIEW':
        sap.ui.getCore().attachInit(function (){
            sap.ushell.Container.createRenderer().placeAt("content")
        });
    break;
    case 'VARIANT':
        sap.ui.getCore().attachInit(function () {
            sap.ushell.Container.attachRendererCreatedEvent(function () {
                sap.ushell.Container.getServiceAsync('AppLifeCycle').then((serviceInstance) => {
                    serviceInstance.attachAppLoaded((event) => {
                        var oView = event.getParameter('componentInstance');
                        sap.ui.require(["sap/ui/rta/api/startAdaptation"], async function (startAdaptation) {
                            var options = {
                                rootControl: oView,
                                validateAppVersion: false,
                                flexSettings: {
                                    layer: '<%- flex.layer %>',
                                    developerMode: <%- flex.developerMode %>
								}
                            };
                            await startAdaptation(option);
                        });
                    })
                });
            });
    break;
    case 'ADAPTATION':
        sap.ui.getCore().attachInit(function () {
            sap.ushell.Container.attachRendererCreatedEvent(function () {
                sap.ushell.Container.getServiceAsync('AppLifeCycle').then((serviceInstance) => {
                    serviceInstance.attachAppLoaded((event) => {
                        var oView = event.getParameter('componentInstance');
                        sap.ui.require(["sap/ui/rta/api/startAdaptation", "<%- flex.pluginModule %>"], async function (startAdaptation, loadPlugin) {
                            var options = {
                                rootControl: oView,
                                validateAppVersion: false,
                                flexSettings: {
                                    layer: '<%- flex.layer %>',
                                    developerMode: <%- flex.developerMode %>
								}
                            };
                            await startAdaptation(options, loadPlugin);
                        });
                    })
                });
            });
        });
    break;
    </script>
```

Reply:
Yes, that makes it easier to maintain. I will do it with the next commit

Reply:
Thanks!

Reply:
Now that I tried it, I am unsure whether it is a good idea or not. I didn't use a switch/case but an  if/if-else/else but I don't think we are gaining anything by having it in three blocks. On the contrary, the amount of duplication compared to the one simplified line seems to me not a good trade-off.
@zdravko-georgiev I am still open to be convinced otherwise though ...

```js
    <script><% if (!locals.flex) { /* simple preview */ %>
        sap.ui.getCore().attachInit(function (){
            <% if (!locals.locateReuseLibsScript) { %>
			sap.ushell.Container.createRenderer().placeAt("content");<% } %>
        });<% } else if (flex.pluginModule) { /* adaptation project */%>
        sap.ui.getCore().attachInit(function () {
            sap.ushell.Container.attachRendererCreatedEvent(function () {
                sap.ushell.Container.getServiceAsync('AppLifeCycle').then((serviceInstance) => {
                    serviceInstance.attachAppLoaded((event) => {
                        var oView = event.getParameter('componentInstance');
                        sap.ui.require(["sap/ui/rta/api/startAdaptation", "<%- flex.pluginModule %>"], async function (startAdaptation, loadPlugin) {
                            var options = {
                                rootControl: oView,
                                validateAppVersion: false,
                                flexSettings: {
                                    layer: '<%- flex.layer %>',
                                    developerMode: <%- flex.developerMode %>
								}
                            };
                            await startAdaptation(options, loadPlugin);
                        });
                    })
                });
            });
            <% if (!locals.locateReuseLibsScript) { %>
			sap.ushell.Container.createRenderer().placeAt("content");<% } %>
        });<% } else { /* variant management */ %>
        sap.ui.getCore().attachInit(function () {
            sap.ushell.Container.attachRendererCreatedEvent(function () {
                sap.ushell.Container.getServiceAsync('AppLifeCycle').then((serviceInstance) => {
                    serviceInstance.attachAppLoaded((event) => {
                        var oView = event.getParameter('componentInstance');
                        sap.ui.require(["sap/ui/rta/api/startAdaptation"], async function (startAdaptation) {
                            var options = {
                                rootControl: oView,
                                validateAppVersion: false,
                                flexSettings: {
                                    layer: '<%- flex.layer %>',
                                    developerMode: <%- flex.developerMode %>
								}
                            };
                            await startAdaptation(options);
                        });
                    })
                });
            });
            <% if (!locals.locateReuseLibsScript) { %>
			sap.ushell.Container.createRenderer().placeAt("content");<% } %>
        });
    <% } %>
    </script>
```

Reply:
OK, let us keep it the way it is now.